gen-resourcesdocs/README.md

```diff
@@ -0,0 +1,124 @@
+# Kubernetes API resources documentation generator
+
+This tool extracts information from the OpenAPI specification file of the [Kubernetes API](https://github.com/kubernetes/kubernetes/blob/master/api/openapi-spec/swagger.json) and creates documentation in Markdown format, suitable for the [Kubernetes website](https://kubernetes.io/docs/reference/kubernetes-api/).
+
+## Outline
+
+The documentation is splitted in several *parts*, and each part contains several *chapters*, each chapter describing a main Kubernetes resource and its associated resources and definitions, followed by the *Operations* operating on the resources documented in the chapter.
+
+The parts and chapters are defined in the `config/<version>/toc.yaml` file.
+
+```yaml
+parts:
+- name: Part I
+  chapters:
+  - name: Deployment
+    group: apps
+    version: v1
+  - name: Pod
+    group: ""
+    version: v1
+    otherDefinitions:
+    - PodSpec
+    - Container
+    - Handler
+    - NodeAffinity
+    - PodAffinity
+    - PodAntiAffinity
+    - Probe
+    - PodStatus
+    - PodList
+- name: Part II
+  chapters:
+  - name: Service
+    group: ""
+    version: v1
+```
+
+In this example, the first part contains two chapters and the second part one chapter.
+
+The first chapter describes the main `Deployment` resource (from the `apps` group and the `v1` version) and its associated resources and definitions. By default, if no `otherDefinitions` are defined, the associated resources are the `List` resource and the `Spec` and `Status` definitions, if appropriate. In this case, `Deployment`, `DeploymentList`, `DeploymentSpec` and `DeploymentStatus` are documented.
+
+The second chapter first describes the main `Pod` resource. The other resources and definitions documented in the chapter are listed in the `otherDefinitions` field.
+
+## Definition Documentation
+
+For each definition (including resources, which are definitions attached to a Group/Version), the fields are listed, with their type and documentation.
+
+### Documentation of composite types
+
+If the type of a field is a composite type, the Definition of this composite type is documented either inline if this Definition is not documented elsewhere, or the type name is a link pointing to the definition of the composite type.
```

Comment:
nit ?: s/Definition/definition/
nit: possibly reword
If the type of a field is a composite type, then the definition of this composite type is documented:
 - inline if not defined in another object, or
 - in another section or page and linked through the type name.

Reply:
> inline if not defined in another object, or

That's not completely true. A definition can be defined several times in different objects. For example, you can see here (https://kubernetes.io/docs/reference/kubernetes-api/workloads-resources/container/#lifecycle) that the `Handler` definition is documented twice, for `lifecycle.postStart` and for `lifecycle.preStop`.

On the other hand, because `ExecAction` is defined here (https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/exec-action/#ExecAction), the type for `lifecycle.postStart.exec` is a lnk to this page.

Reply:
> inline if not defined in another object, or

That's not completely true. A definition can be defined several times in different objects. For example, you can see here (https://kubernetes.io/docs/reference/kubernetes-api/workloads-resources/container/#lifecycle) that the `Handler` definition is documented twice, for `lifecycle.postStart` and for `lifecycle.preStop`.

On the other hand, because `ExecAction` is defined here (https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/exec-action/#ExecAction), the type for `lifecycle.postStart.exec` and `lifecycle.preStop.exec` are links to this page.

Reply:
OK. This may need further clarification on the details.